sdk/ai/Azure.AI.Inference/tests/Samples/Sample8_ChatCompletionsWithTelemetry.cs

```diff
@@ -0,0 +1,360 @@
+            };
+
+            // Call the enpoint and output the response.
+            StreamingResponse<StreamingChatCompletionsUpdate> response = await client.CompleteStreamingAsync(requestOptions);
+            Assert.That(response, Is.Not.Null);
+
+#if !SNIPPET
+            await checkStreamingResponse(response);
+#else
+            await foreach (StreamingChatCompletionsUpdate chatUpdate in response)
+            {
+                if (!string.IsNullOrEmpty(chatUpdate.ContentUpdate))
+                    {
+                        System.Console.Write(chatUpdate.ContentUpdate);
+                    }
+            }
+#endif
+            System.Console.WriteLine("");
+        }
+
+        #region Helpers
+        private async Task checkStreamingResponse(StreamingResponse<StreamingChatCompletionsUpdate> response)
+        {
+            string id = null;
+            string ret_model = null;
+            bool gotRole = false;
+            await foreach (StreamingChatCompletionsUpdate chatUpdate in response)
+            {
+                Assert.That(chatUpdate, Is.Not.Null);
+
+                Assert.That(chatUpdate.Id, Is.Not.Null.Or.Empty);
+                Assert.That(chatUpdate.Created, Is.GreaterThan(new DateTimeOffset(new DateTime(2023, 1, 1))));
+                Assert.That(chatUpdate.Created, Is.LessThan(DateTimeOffset.UtcNow.AddDays(7)));
+                if (!string.IsNullOrEmpty(chatUpdate.Id))
+                {
+                    Assert.That((id is null) || (id == chatUpdate.Id));
+                    id = chatUpdate.Id;
+                }
+                if (!string.IsNullOrEmpty(chatUpdate.Model))
+                {
+                    Assert.That((ret_model is null) || (ret_model == chatUpdate.Model));
+                    ret_model = chatUpdate.Model;
+                }
+                if (chatUpdate.Role.HasValue)
+                {
+                    Assert.IsFalse(gotRole);
+                    Assert.That(chatUpdate.Role.Value, Is.EqualTo(ChatRole.Assistant));
+                    gotRole = true;
+                }
+
+                if (!string.IsNullOrEmpty(chatUpdate.ContentUpdate))
+                {
+                    System.Console.Write(chatUpdate.ContentUpdate);
+                }
+            }
+        }
+
+        private void CheckResponse(Response<ChatCompletions> response)
+        {
+            Assert.That(response, Is.Not.Null);
+            Assert.That(response.Value, Is.InstanceOf<ChatCompletions>());
+            Assert.That(response.Value.Id, Is.Not.Null.Or.Empty);
+            Assert.That(response.Value.Created, Is.Not.Null.Or.Empty);
+            Assert.That(response.Value.Choices, Is.Not.Null.Or.Empty);
+            Assert.That(response.Value.Choices.Count, Is.EqualTo(1));
+            ChatChoice choice = response.Value.Choices[0];
+            Assert.That(choice.Index, Is.EqualTo(0));
+            Assert.That(choice.FinishReason, Is.EqualTo(CompletionsFinishReason.Stopped));
```

Comment:
it should be equal to string "stop", is it?

Reply:
We compare to the Enumeration value defined in Azure.AI.Inference CompletionsFinishReason.Stopped is the synonym of "stop". https://github.com/Azure/azure-sdk-for-net/blob/8945e6ef5e5225306abbd3f62fb7e1b98c44632e/sdk/ai/Azure.AI.Inference/src/Generated/CompletionsFinishReason.cs#L31C8-L31C108